### ISSUE TITLE
model/README.md: vocab.h entry count is wrong (32,768 -> 25,353)

### ISSUE THREAD

**Issue by Bellman281** [Bellman281/esp32-tinyLLM#8]
`model/README.md`'s file table says:

> | `vocab.h` | The tokenizer's token-id → UTF-8 bytes table, 32,768 entries — matches this `model.bin`'s vocab size. Generated data, not logic. |

It has **25,353** entries, and it does *not* match `model.bin`'s vocab size.

## Evidence

`reference-c/esp32-llm-lab/vocab.h`, line 3:

```c
#define VOCAB_N 25353
static const unsigned char VOCAB_BLOB[164388] = {
```

`demo/bpe32768.json` — the tokenizer that produced it — has 25,353 vocab entries. Regenerating the header from it with upstream's `generate_vocab.py` reproduces the shipped `vocab.h` byte-for-byte (`VOCAB_N 25353`, blob `164388` bytes), so the file is not stale.

`model.bin`'s header field `V` is 32,768. Both numbers are real; they describe different things:

| number | what it is |
|---|---|
| 32,768 | rows in the token embedding and the PLE table — the `V` in `V·P·L` = 25M PLE params |
| 25,353 | entries the tokenizer actually has — the only ids that can ever be produced or decoded |

The upstream pipeline asks for a 32,768-token BPE (`prepare.py --vocab 32768`) but the tokenizer it produced stopped at 25,353; the model was still built with `vocab_size=32768`, leaving rows 25353–32767 padded and untrained.

## Why it's worth more than a number swap

The output head computes all 32,768 logits, but no id above 25,352 can ever be emitted. **~23% of the head's work is unreachable.** The head is already the largest single term in the C-vs-Rust gap — 57.1 ms of C's 119.8 ms/token, and +33.2 ms of the +59.8 ms absolute gap (`BENCHMARKING.md`).

That's not something to act on in this issue, but the doc shouldn't state a number that hides it. Upstream's newer export format carries `output_vocab` separately from `input_vocab` for exactly this reason — see the tracking issue on other models, and `TRAINING.md`'s "The export format gap".

## Fix

Correct the count, and add a short section stating both numbers explicitly so the next reader doesn't have to re-derive which is which.

Cosmetic, but this repo's whole premise is that the documented numbers are the measured ones.
